### PR TITLE
minor graph middleware refactor, no logic changes

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #2951

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Improve restore time on large restores by optimizing how items are loaded from the remote repository.
 - Remove exchange item filtering based on m365 item ID via the CLI.
 - OneDrive backups no longer include a user's non-default drives.
+- Refined oneDrive rate limiter controls to reduce throttling errors.
```
Comment (pandeyabs) [inline CHANGELOG.md:20]: nit: OneDrive*

**Review by pandeyabs — COMMENTED** [review]
Please rebase this too. I'll review both of them after that. Thanks!

File: src/internal/connector/data_collections.go [hunk 2]
```diff
@@ -213,15 +217,16 @@ func (gc *GraphConnector) ConsumeRestoreCollections(
 		return nil, clues.Wrap(err, "malformed azure credentials")
 	}
 
-	switch selector.Service {
+	switch sels.Service {
 	case selectors.ServiceExchange:
+		ctx = graph.BindRateLimiterConfig(ctx, graph.LimiterCfg{Service: path.ExchangeService})
```
Comment (pandeyabs) [inline src/internal/connector/data_collections.go:222]: @ryanfkeepers rebase didn't go through it seems? This change is from an older PR.

File: src/internal/connector/graph/concurrency_middleware.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
+package graph
```
Comment (pandeyabs) [inline src/internal/connector/graph/concurrency_middleware.go:1]: nit: `concurrency_middleware.go` name is slightly misleading, given we have also added rate limiter to this file.  `limiter_middleware.go` maybe?

**Review by pandeyabs — APPROVED** [review]
LGTM!

File: src/internal/connector/graph/concurrency_middleware.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/connector/graph/concurrency_middleware.go:1]: `limiter` is equally generic.  What does it limit exactly?  Dinosaurs, perhaps...

Two-way door.  If one of us comes up with a great name, we can always change it up in a follow-up.

**Pull request merged**